src/core/control/settings/Settings.cpp

```diff
@@ -224,6 +229,42 @@ void Settings::loadDefault() {
     /**/
 }
 
+auto Settings::loadViewMode(size_t mode) -> bool {
+    if (mode < 0 || mode > viewModeAttributes.size()) {
+        return false;
+    }
+    auto attributes = viewModeAttributes.at(mode);
+    menubarVisible = false;
+    showSidebar = false;
+    showToolbar = false;
+    fullscreenActive = false;
+    for (const string& attr: StringUtils::split(attributes, ',')) {
+        if (attr == "showMenubar") {
+            menubarVisible = true;
+        } else if (attr == "showSidebar") {
+            showSidebar = true;
+        } else if (attr == "showToolbar") {
+            showToolbar = true;
+        } else if (attr == "goFullscreen") {
+            fullscreenActive = true;
+        }
+    }
+    this->activeViewMode = mode;
+    return true;
+}
+
+auto Settings::getActiveViewMode() const -> bool {
+    return activeViewMode;
+}
+
+auto Settings::getViewModeStrings() const -> std::vector<std::string> {
```

Comment:
Return a const ref to avoid unecessary copies

```suggestion
auto Settings::getViewModeStrings() const -> const std::vector<std::string>& {
```